projects/plugins/jetpack/extensions/shared/memberships/settings.js

```diff
@@ -255,12 +165,13 @@ export function NewsletterAccessDocumentSettings( { accessLevel, setPostMeta } )
 				isLoading: isApiStateLoading(),
 				stripeConnectUrl: getConnectUrl(),
 				hasNewsletterPlans: getNewsletterProducts()?.length !== 0,
-				hasPaywallBlock: getBlocks().some( block => block.name === 'jetpack/paywall' ),
+				foundPaywallBlock: getBlocks().find( block => block.name === paywallBlockName ),
```

Comment:
Nit; I did like the previous `hasPaywallBlock` as a const name little better, but either works. :-) Maybe even `postHasPaywallBlock` to be extra clear?